src/shared/orgOpenCommandBase.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import path from 'node:path';
+import fs from 'node:fs';
+import { platform, tmpdir } from 'node:os';
+import { execSync } from 'node:child_process';
+import isWsl from 'is-wsl';
+import { apps } from 'open';
+import { SfCommand } from '@salesforce/sf-plugins-core';
+import { Connection, Messages, Org, SfdcUrl, SfError } from '@salesforce/core';
+import { env, sleep } from '@salesforce/kit';
+import utils, { fileCleanup, getFileContents, handleDomainError } from './orgOpenUtils.js';
+import { type OrgOpenOutput } from './orgTypes.js';
+
+Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
+const messages = Messages.loadMessages('@salesforce/plugin-org', 'open');
+
+type OrgOpenFlags = {
+  'url-only': boolean;
+  browser?: 'chrome' | 'firefox' | 'edge';
+  path?: string;
+  private: boolean;
+};
+
+export abstract class OrgOpenCommandBase<T> extends SfCommand<T> {
```

Comment:
my preference is not use base command classes (it can have some surprising side-effects like loosing the value of the static `enableJsonFlag` prop).

Especially since there's only one method here that's being shared